jax/_src/scipy/signal.py

```diff
@@ -33,6 +33,68 @@
 from jax._src.typing import Array, ArrayLike
 from jax._src.util import canonicalize_axis, tuple_delete, tuple_insert
 
+@_wraps(osp_signal.fftconvolve)
+def fftconvolve(in1: ArrayLike ,in2: ArrayLike, mode: str ="same",axes=None) -> Array:
+  # sanity checks
+  check_arraylike('fftconvolve', in1, in2)
+  in1, in2 = promote_dtypes_inexact(in1, in2)
+  if in1.ndim == in2.ndim == 0:  # scalar inputs
+    return in1 * in2
+  elif in1.ndim != in2.ndim:
+    raise ValueError("in1 and in2 should have the same dimensionality")
+  elif in1.size == 0 or in2.size == 0:  # empty arrays
+    return jnp.array([], dtype=in1.dtype)
+  # warn current limitations
+  if mode != "same":
+    raise NotImplementedError("current fftconvolve() uses only mode=same argument")
+  if axes is not None:
+    raise NotImplementedError("current fftconvolve() does not use axes argument")
+
+  s1 = in1.shape
+  s2 = in2.shape
+
+  axes = range(in1.ndim)
+  # Axes of length 1 can rely on broadcasting rules for multipy,
+  # no fft needed.
+  axes = [a for a in axes if s1[a] != 1 and s2[a] != 1]
+
+  try:
+    jnp.broadcast_shapes(tuple(np.delete(np.array(s1),axes)), 
+      tuple(np.delete(np.array(s2),axes)))
+  except ValueError as err:
+    raise ValueError('in1 and in2 cannot be broadcast together.') from err
+
+  shape = [s1[i] + s2[i] - 1 for i in range(in1.ndim)]
+
+  # crude next_fast_len equivalent with just power of 2
+  def _next_fast_len(n):
+    return 2**np.ceil(np.log2(n)).astype('uint16')
+
+  fshape = [_next_fast_len(shape[a]) for a in axes]
+
+  complex_result = (in1.dtype.kind == 'c' or in2.dtype.kind == 'c')
```

Comment:
`in1` and `in2` have the same dtype at this point, so I'd probably do
```
complex_result = jnp.iscomplexobj(in1)
```
And since it's only used once, you could call that directly in the `if` statement

Reply:
Yes ! with the promote_dtypes_inexact it is ok, and seems to work even with mixture of complexe & flaot, incredible :)